cookbook/oled.rst

```diff
@@ -0,0 +1,273 @@
+A guide to using SSD1306 OLED display
+=====================================
+
+All the info in this guide can be found in the documentation :doc:`ssd1306_i2c </components/display/ssd1306_i2c>`
+and :doc:`Display </components/display/>`.
+This article will show some practical examples on how to use it, and maybe it can help beginners getting started
+using this display. The documentation will have more in depth info on all the options available.
+These displays are pretty cheap on
+`AliExpress <https://www.aliexpress.com/item/10pcs-0-96-yellow-blue-0-96-inch-OLED-module-New-128X64-OLED-LCD-LED-Display/32638669209.html>`__
+and they are quite bright.
+There is also a 1.3" version with same resolution that should work with the same config as this one, however I have not tested them personally yet.
+
+.. note::
+
+    The display used in the example pictures is a dual color display. X 0-127 through Y 0-15 are yellow pixels, the rest are blue.
+    There are different versions of the display.
+
+The display itself is a 0.96” OLED unit that connects to your ESP via ``i2c``, you will need to add ``i2c`` configuration to your node.
+It has 4 pins that needs to be connected: ``GND``, ``VCC`` (3.3v), ``SCL`` and ``SDA`` (some of them have a misprint and the SCL pin says SCK for some reason)
+
+.. figure:: images/oled-xy.png
+    :align: center
+    :width: 80.0%
+
+The XY orientation of the display.
+
+Configuration
+_____________
+i2c:
+****
+
+If you don't know the i2c address for your display:
+
+.. code-block:: yaml
+
+    i2c:
+      sda: D5
+      scl: D6
+      scan: true
+
+Set ``scan: true`` if you don’t know the address of your display and check the log for the node:
+
+.. code-block::
+
+    [11:54:53][C][i2c:043]: I2C Bus:
+    [11:54:53][C][i2c:044]:   SDA Pin: GPIO14
+    [11:54:53][C][i2c:045]:   SCL Pin: GPIO12
+    [11:54:53][C][i2c:046]:   Frequency: 50000 Hz
+    [11:54:53][I][i2c:048]: Scanning i2c bus for active devices...
+    [11:54:53][I][i2c:055]: Found i2c device at address 0x3C
+
+Take note of the address and set ``scan: false`` again.
+(The board used in this example is a Wemos D1 mini, so adapt pins to the board you are using)
+
+Node config:
+************
+
+The configuration for your node should look something like this:
+(your basic node configuration, e.g. network and such is not discussed in this guide):

```

Comment:
Reverse this, first everything should be roughly explained, and at the end the full config will follow.